api/post.go

```diff
@@ -776,11 +780,34 @@ func sendNotificationEmail(post *model.Post, user *model.User, channel *model.Ch
 	if channel.Type == model.CHANNEL_DIRECT {
 		bodyText = userLocale("api.post.send_notifications_and_forget.message_body")
 		subjectText = userLocale("api.post.send_notifications_and_forget.message_subject")
-		channelName = senderName
+
+		if result := <-Srv.Store.Preference().Get(user.Id, model.PREFERENCE_CATEGORY_DISPLAY_SETTINGS, "name_format"); result.Err != nil {
+			channelName = senderName
+		} else {
+			if result.Data.(model.Preference).Value == "nickname_full_name" {
```

Comment:
You can actually use `User.GetDisplayNameForPreference(nameFormat)` here instead of needing this whole if-else block.

Can you also add a null check for `sender` here? We shouldn't ever pass a null sender into the function since there's no system messages that are sent over direct channels, but it would be nice to have just in case those are added.

Other than that, everything else looks good. Thanks!